### ISSUE TITLE
Missed mis-indent

### ISSUE THREAD

**Issue by smurfix** [PyCQA/pylint#3660]
### Steps to reproduce
This code has an obvious mis-indented line which pylint doesn't notice.
```
"""test.py"""
def main():
    """some code"""
    def runner():
        pass  # something interesting
        runner()
main()
```
I found this in a test runner which caused a bunch of tests in our code to silently (and, needless to say, mistakenly) get marked as OK.

### Current behavior
No warning.

### Expected behavior
Two warnings:
* `main()` does not contain any actual code
* `runner()` is unused. (The "recursive" call inside `runner()` itself should be ignored.)

### pylint --version output
```
pylint 2.4.4
astroid 2.3.3
Python 3.8.3 (default, May 14 2020, 11:03:12) 
```

**Comment by Pierre-Sassoulas** [comment]
I'm going to close since the issue was opened we have ``unnecessary-pass`` that would permit to detect that something is wrong with the code. I think other possible check on this code would have a lot of false positives.
